src/fidesapi/routes/generate.py

```diff
@@ -0,0 +1,153 @@
+"""
+Contains all of the endpoints required to manage generating resources.
+"""
+from enum import Enum
+from typing import Dict, List, Optional, Union
+
+from fastapi import APIRouter, HTTPException, Response, status
+from fideslang.models import Organization, System
+from loguru import logger as log
+from pydantic import BaseModel
+
+from fidesapi.routes.crud import get_resource
+from fidesapi.routes.util import API_PREFIX, unobscure_string
+from fidesapi.sql_models import sql_model_map
+from fidesctl.core.system import generate_aws_systems
+
+
+class GenerateTypes(str, Enum):
+    """
+    Generate Type Enum to capture the discrete possible values
+    for a valid type of resource to generate.
+    """
+
+    SYSTEM = "systems"
+    DATASET = "datasets"
+
+
+class AWSConfig(BaseModel):
+    """
+    The model for the connection config for AWS
+    """
+
+    region_name: str
+    aws_secret_access_key: str
+    aws_access_key_id: str
+
+
+class OktaConfig(BaseModel):
+    """
+    The model for the connection config for Okta
+    """
+
+    okta_client_token: str
+
+
+class Generate(BaseModel):
+    """
+    Defines attributes for generating resources included in a request.
+    """
+
+    config: Union[AWSConfig, OktaConfig]
+    target: str
+    type: GenerateTypes
+
+
+class GenerateRequestPayload(BaseModel):
+    """
+    The model for the request body housing generate information.
+    """
+
+    organization_key: str
+    generate: Generate
+
+
+class GeneratedResponse(BaseModel):
+    """
+    The model to hous the response for generated infrastructure.
+    """
+
+    generate_results: Optional[List[System]]
```

Comment:
What will this model look like if we also add datasets? Will there be different collections for systems and datasets?

Reply:
I think (based on a recent change from @brentonmallen1 actually) we would either need to let it be very loose (`generate_results: Optional[List]`) or define them in separate models as their respective `List` object and union those models together

Reply:
Do we envision that this api could be used for more than one scan call at at time? If so then we might just want a field per type so that the client knows which is which. I can also imagine a scenario where scanning okta or aws returns systems and datasets and we drop the concept of scanning for either exclusively.

Reply:
updated to include `Dataset` as well - I don't think we need to be as specific as originally thought with the types here since they are defined (tested ok anyways)